src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/StatusServiceManager.java

```diff
@@ -17,4 +31,33 @@ public boolean canBeUnlisted(Statusable entity) {
     public boolean shouldBeCountInUtilization(Statusable entity) {
         return entity.getStatus().isCountInUtilization();
     }
+
+    public PossibleActions getPossibleActions(Hearing hearing) {
+        PossibleActions possibleActions = new PossibleActions();
+
+        possibleOperationConfig.stream()
+            .filter(validator -> (Boolean) validator.getDbConfigVerifier().apply(hearing))
+            .forEach(validator -> validator.getBusinessOperationVerifier().apply(hearing, possibleActions));
+
+        return possibleActions;
+    }
+
+    private static Function<Statusable, Boolean> checkIfStatusCanBeAdjourned = (Statusable entity) ->
+        entity.getStatus().isCanBeAdjourned();
+
+    private static Function<Statusable, Boolean> checkIfStatusCanBeWithdrawn = (Statusable entity) ->
```

Comment:
Can this response be adapted to allow for these cases?

4. Option to Withdraw is not available if the request is listed and today's date is > the start date of the hearing i.e.
> always allowed for unlisted requests (already done)
> allowed for listed requests only if they haven't passed their start date

(Start date of hearing for a multi-session request is the date of the earliest session into which one of its hearing parts is listed)